Exercise_3.java

```diff
@@ -17,31 +17,48 @@
         // Constructor 
         Node(int d) 
         { 
+        	this.data = d;
             //Write your code here 
         } 
-    } 
+    }
   
     // Method to insert a new node 
     public static LinkedList insert(LinkedList list, int data) 
     { 
+    	Node node =new Node(data);
         // Create a new node with given data 
    
         // If the Linked List is empty, 
-        // then make the new node as head 
-        
+    	if(list.head==null)

```

Comment:
Good Job! Please add comments and time and space complexity for all the operations